nixos/modules/services/hardware/amdgpu.nix

```diff
@@ -0,0 +1,43 @@
+{ config, lib, pkgs, ... }:
+
+let
+  cfg = config.hardware.amdgpu;
+in {
+  options.hardware.amdgpu = {
+    legacySupport.enable = lib.mkEnableOption ''
+      using `amdgpu` kernel driver instead of `radeon` for Southern Islands
+      (Radeon HD 7000) series and Sea Islands (Radeon HD 8000)
+      series cards. Note: this removes support for analog video outputs,
+      which is only available in the `radeon` driver
+    '';
+    initrd.enable = lib.mkEnableOption ''
+      loading `amdgpu` kernelModule in stage 1.
+      Can fix lower resolution in boot screen during initramfs phase
+    '';
+    opencl.enable = lib.mkEnableOption ''OpenCL support'';
+    # cfg.amdvlk option is defined in ./amdvlk.nix module
+  };
+
+  config = {
+    boot.kernelParams = lib.optionals cfg.legacySupport.enable [
+      "amdgpu.si_support=1"
+      "amdgpu.cik_support=1"
+      "radeon.si_support=0"
+      "radeon.cik_support=0"
+    ];
+
+    boot.initrd.kernelModules = lib.optionals cfg.initrd.enable [ "amdgpu" ];
+
+    hardware.opengl = lib.mkIf cfg.opencl.enable {
+      enable = lib.mkDefault true;
+      extraPackages = [
+        pkgs.rocmPackages.clr
+        pkgs.rocmPackages.clr.icd
```

Comment:
I don't know the current landscape around OpenCL but there's also rusticl which might actually be preferable?

The option description should at least mention that this activates the ROCm OpenCL packages.

Reply:
I have no idea if `rusticl` by Mesa even works.

But sure, "rocm opencl" done.

Reply:
Perhaps we should namespace the option under `hardware.amdgpu.rocm.opencl.enable`. There will likely be more ROCM-related options in the future.

Reply:
I think we can use `mkRenamedOption` then, if someone decides to add more options under the ROCM.

Let's not clutter the PR and expand the scope beyond what's it supposed to.